AdobeStockClient/Model/SearchParametersProviderPool.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Magento\AdobeStockClient\Model;
+
+use AdobeStock\Api\Models\SearchParameters;
+use Magento\AdobeStockClientApi\Api\SearchParameterProviderInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+
+class SearchParametersProviderPool implements SearchParameterProviderInterface
```

Comment:
I called it pool to clarify how this class should be used. It's not expected to have multiple levels of subsequent classes for applying criteria to search parameters.

In order to avoid confusion, I'd keep it a `Pool`

Reply:
But its usage shows it as composite because you call the interface and configure the preference via di.xml as composite, not as a pool. As for me, the `Pool` more confusing name rather than `Composite`, what signs of the pool this class has?